authors/apps/articles/views.py

```diff
@@ -91,3 +91,88 @@ def update(self, request, slug,*args,**kwargs):
         serializer.is_valid(raise_exception=True)
         serializer.save()
         return Response(serializer.data)
+
+
+class ShareFacebookView(APIView):
+    """creating links for sharing articles via facebook
+    """
+
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (ArticleShareLinkRenderer, )
+
+    def get_object(self, *args, **kwargs):
+        slug = self.kwargs.get("slug")
+        return get_object_or_404(Article, slug=slug)
+
+    def post(self, request, slug, *args, **kwargs):
+        self.get_object(slug)
+        current_url = 'https://{}'.format(get_current_site(request))
+        route = 'api/articles'
+
+        url = "{}/{}/{}".format(current_url, route, slug)
+
+        facebook_url = "https://www.facebook.com/sharer/sharer.php?u="
+        message = {"Facebook_link": facebook_url + url}
+
+        return Response(message)
+
+
+class ShareTwitterView(APIView):
+    """creating links for sharing articles via facebook
+    """
+
+    permission_classes = (IsAuthenticated,)
+    renderer_classes = (ArticleShareLinkRenderer, )
+
+    def get_object(self, *args, **kwargs):
+        slug = self.kwargs.get("slug")
+        return get_object_or_404(Article, slug=slug)
+
+    def post(self, request, slug, *args, **kwargs):
+        self.get_object(slug)
+        current_url = 'https://{}'.format(get_current_site(request))
+        route = 'api/articles'
+
+        url = "{}/{}/{}".format(current_url, route, slug)
+
+        twitter_url = "https://twitter.com/home?status="
+        message = {"twitter_link": twitter_url + url}
+
+        return Response(message)
+
+
+class ShareEmailView(ListCreateAPIView):
```

Comment:
@frankopkusianwar, use the CreateAPIView.it has only one method to `post`
The `ListCreateAPIView` is raising errors when I attempt to send a get